### ISSUE TITLE
Allow users to set username after first login with external provider

### ISSUE THREAD

**Issue by ratte1303** [zorn-v/nextcloud-social-login#445]
At the moment the usernames are cryptic with a prefix or in the case of telegram an numeric id. 

Is it possible for the user to set their nextcloud username after sucessfull authentication with a 3rd oarty provider via a simple form. So the username is not copied from the external provider but is being set by the user?

:)

**Comment by blindzero** [comment]
That would be so awesome! Especially, as usernames are not able to change the username later.

**Comment by zorn-v** [comment]
Username (uid) can not be changed, tt is nextcloud restriction. Change display name.

**Comment by blindzero** [comment]
@zorn-v 
Yes and no... User creation by admin as well as self-registration allows that.
So it should be possible to influence that on creation. Whether by configuration, eg mapping a scope field or, by dialogue?

**Comment by zorn-v** [comment]
OAuth is not trusted source, so uid only with prefix from there.

**Comment by blindzero** [comment]
@zorn-v 
Sorry, doesn't really make sense...self-registration is also no trusted source and there it works...
But...your project, your decision.  Just a pitty that demand of users is not seen...

**Comment by zorn-v** [comment]
There is no separate "registration" in my app. If user with provided uid exists, it just login as that user. 
So if oauth return something like "admin" you login as admin (same with other existing users).

**Comment by zorn-v** [comment]
Sorry, did not correct understand original feature request. 
It is exactly about separate registration step.
Maybe later

**Comment by rinuwaii** [comment]
This would be really nice. I'd like to use this to allow my private development Discord server to access my Nextcloud, but I'm worried if, for example, their Discord accounts get terminated randomly (as happens with the state of Discord support) they lose access to their profile. 

Adding in a registration step where they assign themselves a username and then they can either login with that or their Discord account (as well as unlink and re-link their account if needed) would be great.

**Comment by zorn-v** [comment]
> they lose access to their profile

They can set password via "forgot password" for example, and login as usual. 
Did not understand how setting username helps with deleted account in discord.

**Comment by autra** [comment]
One more use  case : it'll be easier for them to setup caldav / carddav connexion afterwards.
